### PR TITLE
#160584009-Pagination support for articles

### PR DESCRIPTION
**Pull request by j0flintking02**
#### What does this PR do?
    Articles should have pagination support
    
#### Description of Task to be completed?
    - This is to allow a specific number of articles on the page (10)
    - This is to allow easy access to articles while the user is trying to view  the articles

#### How should this be manually tested?
- activate the virtual environment
```console
$ virtualenv env
```
- activate the virtual environment
```console
$ source env/bin/activate
```
- install the projects dependences
```console
$ pip install -r requirments.txt
```
in the terminal run the migration
```console
 $ python manage.py makemigrations articles
 $ python manage.py makemigrations authentication
 $ python manage.py makemigrations profiles
 $ python manage.py migrate
 ``` 
- One can be to view only 10 articles per page
   > api/articles/ [GET]
```json
{
    "count": 3,
    "next": null,
    "previous": null,
    "results": [
        {
            "slug": "how-to-train-your-dragons",
            "title": "How to train your dragons",
            "description": "Ever wonder how?",
            "body": "You have to believe",
            "tagList": [
                "reactjs",
                "angularjs",
                "dragons"
            ],
            "createdAt": "2018-10-15T16:51:46.904456Z",
            "updatedAt": "2018-10-15T16:51:46.904477Z",
            "favorited": false,
            "favoritesCount": 0,
            "rating": 0,
            "ratingsCount": 0,
            "author": 1
        },
        {
            "slug": "how-to-train-your-dragon00",
            "title": "How to train your dragon",
            "description": "Ever wonder how?",
            "body": "You have to believe",
            "tagList": [
                "reactjs",
                "angularjs",
                "dragons"
            ],
            "createdAt": "2018-10-15T17:41:27.965130Z",
            "updatedAt": "2018-10-15T17:41:27.965150Z",
            "favorited": false,
            "favoritesCount": 0,
            "rating": 0,
            "ratingsCount": 0,
            "author": 1
        },
        {
            "slug": "how-to-train-your-dragonss",
            "title": "How to train your dragonss",
            "description": "Ever wonder how?",
            "body": "You have to believe",
            "tagList": [
                "reactjs",
                "angularjs",
                "dragons"
            ],
            "createdAt": "2018-10-15T17:41:34.822289Z",
            "updatedAt": "2018-10-15T17:41:34.822309Z",
            "favorited": false,
            "favoritesCount": 0,
            "rating": 0,
            "ratingsCount": 0,
            "author": 1
        }
    ]
}
```
### What are the relevant PT stories
[#160584009](https://www.pivotaltracker.com/n/projects/2198584/stories/160584009)

### CODE REVIEW

**Comment by Edward-K1** [comment]
Good work, @j0flintking02, but please include screenshots of you testing the endpoints using postman.
Also, please improve this 
image

**And correct this 👇  (you're talking about comments instead of articles)**
image

**Review by balaakagordon — APPROVED** [review]
Great work. Thanks for taking the time to refactor the codebase. But I agree with Edward's comment. Try to include images from Postman which show how the feature has been manually tested

**Comment by JoyLubega** [comment]
Rebase

**Review by kenneth051 — COMMENTED** [review]
Good work Jonathan,  I have gone through the code and am impressed.

File: authors/apps/articles/mixins.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class MyPaginationMixin(object):
```
Comment (nakimera) [inline authors/apps/articles/mixins.py]: Please change this class name to `AhPaginationMixin`.

**Comment by balaakagordon** [comment]
LGTM

**Review by Edward-K1 — APPROVED** [review]
LGTM

**Comment by kenneth051** [comment]
LGTM

**Pull request merged**